src/DynamoCore/Search/SearchDictionary.cs

```diff
@@ -342,7 +343,7 @@ internal void RebuildTagDictionary()
                             tagAndWeight =>
                                 new
                                 {
-                                    Tag = tagAndWeight.Key,
+                                    Tag = tagAndWeight.Key.Substring(0, tagAndWeight.Key.Length > LIMIT_SEARCH_TAG_SIZE ? LIMIT_SEARCH_TAG_SIZE : tagAndWeight.Key.Length),
```

Comment:
is this actually useful at this point? Does the cost of all these extra conditional undo the potential savings?

Reply:
Same q here.

Reply:
well...after all search tags have been added/removed  `RebuildTagDictionary` is called only once.
Performance penalty here should be negligible, compared to the constant searches on each text change.